addon/model/readers/selection-reader.ts

```diff
@@ -51,68 +54,70 @@ export default class SelectionReader implements Reader<Selection, ModelSelection
    * Convert a DOM position to a {@link ModelPosition}
    * Can be null when the {@link Selection} is empty.
    * @param container
-   * @param offset
+   * @param domOffset
    */
-  readDomPosition(container: Node, offset: number): ModelPosition | null {
-    const {container: normalizedContainer, offset: normalizedOffset} = this.normalizeDomPosition(container, offset);
-    const modelNode = this.model.getModelNodeFor(normalizedContainer);
-    const root = this.model.rootModelNode;
-    if (!modelNode) {
-      return null;
+  readDomPosition(container: Node, domOffset: number): ModelPosition | null {
+    let rslt = null;
+    if(TEXT_PROPERTY_NODES.has(tagName(container))) {
+      return this.findPositionForTextPopertyNode(container, domOffset);
     }
-    return ModelPosition.fromParent(root, modelNode, normalizedOffset);
-  }
 
-  /**
-   * Things are easier to work with if we use textnodes as parents as much as possible. This tries
-   * to convert a dom position anchored to an element into an equivalent position anchored to a textnode, but in
-   * a conservative way.
-   * @param container
-   * @param offset
-   * @private
-   */
-  private normalizeDomPosition(container: Node, offset: number): { container: Node, offset: number } {
-    if (isTextNode(container)) {
-      return {container, offset};
+    else if(isElement(container)) {
+      const modelContainer = this.model.getModelNodeFor(container) as ModelElement;
+      const basePath = modelContainer.getOffsetPath();
+
+      const finalOffset = modelContainer.indexToOffset(domOffset);
+      basePath.push(finalOffset);
+      rslt = ModelPosition.from(modelContainer.root, basePath);
+
+
+    } else if (isTextNode(container)) {
+      const modelTextNode = this.model.getModelNodeFor(container) as ModelText;
+      const modelContainer = modelTextNode.parent!;
+      const basePath = modelContainer.getOffsetPath();
+
+      const finalOffset = modelTextNode.getOffset() + domOffset;
+      basePath.push(finalOffset);
+      rslt = ModelPosition.from(modelContainer.root, basePath);
+
     }
-    // try to find a textnode to the left
-    if (offset > 0) {
-      let leftNode: ChildNode | null = container.childNodes[offset - 1];
-      // ignore breaks to the left
-      // TODO: what about other block elements? Do we actually want to do this here, or should we expect
-      // commands to handle breaks themselves?
-      while (leftNode && tagName(leftNode) === "br") {
-        leftNode = leftNode.previousSibling;
+    return rslt;
+  }
+  private findPositionForTextPopertyNode(container: Node, domOffset: number): ModelPosition {
+    if(container.childNodes.length === 0) {
+      // this is fallback behavior in case the dom does something really weird
```

Comment:
not sure if this can never happen, are we assuming container is an element and excluding void elements here?

Reply:
this function is only called when container is a textpropertynode, i.e.. <strong>, <em>, etc